src/background/services/openPayments.ts

```diff
@@ -326,13 +326,68 @@ export class OpenPaymentsService {
       assetScale: walletAddress.assetScale
     })
 
+    await this.initClient(walletAddress.id)
+    await this.completeGrant(amount, walletAddress, recurring)
+
+    await this.storage.set({
+      walletAddress,
+      rateOfPay,
+      minRateOfPay,
+      maxRateOfPay,
+      connected: true
+    })
+  }
+
+  async addFunds({ amount, recurring }: AddFundsPayload) {
+    const { walletAddress } = await this.storage.get(['walletAddress'])
+
+    // clear existing grants of same type, if any
+    const grants = await this.storage.get(['oneTimeGrant', 'recurringGrant'])
+    if (grants.oneTimeGrant && !recurring) {
+      await this.cancelGrant(grants.oneTimeGrant.continue)
+      await this.storage.set({
+        oneTimeGrant: null,
+        oneTimeGrantSpentAmount: '0'
+      })
+    } else if (grants.recurringGrant && recurring) {
+      await this.cancelGrant(grants.recurringGrant.continue)
+      await this.storage.set({
+        recurringGrant: null,
+        recurringGrantSpentAmount: '0'
+      })
+    }
```

Comment:
Let's move this after we call `completeGrant`. In case something fails we are not going to clear the existing grants.

Reply:
`completeGrant` sets these in storage, hmm. Yeah we just need to cancel grants then.